pkg/asset/agent/joiner/clusterinfo.go

```diff
@@ -204,10 +204,13 @@ func (ci *ClusterInfo) retrieveProxy() error {
 	if err != nil {
 		return err
 	}
+	// Remove any duplicates in noProxy
+	noProxy := ci.removeDuplicates(proxy.Spec.NoProxy)
```

Comment:
q: any specific reason for checking only the NoProxy? Could make sense to remove any duplication from all the Proxy fields?